charmcraft/config.py

```diff
@@ -262,6 +262,35 @@ def validate_charm_type(cls, charm_type, values):
             raise ValueError("must be either 'charm' or 'bundle'")
         return charm_type
 
+    @classmethod
+    def expand_short_form_bases(cls, bases: List[Dict[str, Any]]) -> None:
+        """Expand short-form base configuration into long-form in-place."""
+        for index, base in enumerate(bases):
+            # Skip if already long-form. Account for common typos in case user
+            # intends to use long-form, but did so incorrectly (for better
+            # error message handling).
+            if (
+                "run-on" in base
+                or "run_on" in base
+                or "build-on" in base
```

Comment:
Alternative way of writing this, if you like:

```
if {"run-on", "run_on", "build-on", "build_on"} & set(base):
    continue
```

Reply:
interesting, though I will keep it as-is for readability.

another succinct option could have been:
`any(k in base for k in ["build-on", "build_on", "run-on", "run_on"])`